commitizen/commands/commit.py

```diff
@@ -110,17 +110,18 @@ def __call__(self):
         if dry_run:
             raise DryRunExit()
 
-        signoff: bool = (
-            self.arguments.get("signoff") or self.config.settings["always_signoff"]
-        )
+        always_signoff: bool = self.config.settings["always_signoff"]
+        signoff: bool = self.arguments.get("signoff")
+
+        extra_args = self.arguments.get("extra_cli_args", "")
 
         if signoff:
             out.warn(
                 "signoff mechanic is deprecated, please use `cz commit -- -s` instead."
             )
-            extra_args = self.arguments.get("extra_cli_args", "--") + " -s"
-        else:
-            extra_args = self.arguments.get("extra_cli_args", "")
+
+        if always_signoff or signoff:
+            extra_args = f"{extra_args} -s".strip()
```

Comment:
What would happen if `-s` already in extra_args?

Reply:
It's fine, tested with `git cz commit -s -- -s -s` :grin: